test/test_array_ops.cpp

```diff
@@ -81,6 +81,50 @@ TEST(ArrayOps, PreventGradient) {
   }
 }  // end of op PreventGradient
 
+// Test op: Shape, outputs the shape of a tensor
+TEST(ArrayOps, Shape2D) {
+  Scope root = Scope::NewRootScope();
+  int dim1 = 2;
+  int dim2 = 2;
+
+  Tensor A(DT_FLOAT, TensorShape({dim1, dim2}));
+
+  AssignInputValues(A, 7.5f);
+
+  vector<int> static_input_indexes = {};
+  auto R = ops::Shape(root, A);
+
+  vector<DataType> output_datatypes = {DT_INT32};
+
+  std::vector<Output> sess_run_fetchoutputs = {R};
+  OpExecuter opexecuter(root, "Shape", static_input_indexes, output_datatypes,
+                        sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}  // end of op Shape2D
+
+TEST(ArrayOps, Shape3D) {
+  Scope root = Scope::NewRootScope();
+  int dim1 = 2;
+  int dim2 = 3;
+  int dim3 = 4;
+
+  Tensor A(DT_FLOAT, TensorShape({dim1, dim2, dim3}));
+
+  AssignInputValues(A, 7.5f);
+
+  vector<int> static_input_indexes = {};
+  auto R = ops::Shape(root, A);
+
+  vector<DataType> output_datatypes = {DT_INT32};
+
+  std::vector<Output> sess_run_fetchoutputs = {R};
+  OpExecuter opexecuter(root, "Shape", static_input_indexes, output_datatypes,
+                        sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
```

Comment:
Shape2D and Shape3D functions could have been fused into one single function looping over a vector of possible shapes (like Sindhu's fill).

Reply:
I like to keep them separate since that gives more readability. For this test though, I'll only have the 3D - forgot to get rid of the 2D.

Reply:
Keeping both 2D and 3D since 2D tests for Int64 and 3D for Int32

Reply:
looks good here